Add thumbnails for workflow templates

christian-byrne opened the pull request:
Adds thumbnails for templates workflows, following up on https://github.com/Comfy-Org/ComfyUI_frontend/pull/2541. There are different component variants for thumbnails: audio, zoom on hover, dissolve on hover, image compare slider. The variant is specified per template in `coreTemplates.ts`.
 
https://github.com/user-attachments/assets/a565f518-b457-4e30-9144-672a7b5e8599

┆Issue is synchronized with this [Notion page](https://www.notion.so/PR-2729-Add-thumbnails-for-workflow-templates-1a66d73d3650818fb267fc759507b07c) by [Unito](https://www.unito.io)

christian-byrne committed:
create thumbnail components

christian-byrne committed:
remove duplicate translation

christian-byrne committed:
update type for thumbnail variant field

christian-byrne committed:
add media with embedded wfs and stripped metadata

christian-byrne committed:
remove old thumbnails

christian-byrne committed:
convert jpg to png

christian-byrne committed:
add new sd3.5 workflows

christian-byrne committed:
add test verifying files

christian-byrne committed:
adjust old test for changed interaction

christian-byrne committed:
refactor

christian-byrne committed:
remove duplicate image

Commit:
Update locales [skip ci]

huchenlei commented on src/components/templates/thumbnails/CompareSliderThumbnail.vue:
PrimeVue has a image compare component. Is there any reason why we don't use it? https://primevue.org/imagecompare/

christian-byrne replied:
I used it initially, but the slider element and state are not exposed in the [API](https://primevue.org/imagecompare/#api.imagecompare) and it it doesn't have any border between the two images, making it hard to tell that the slider is even there. Based on research of workflow and model sharing sites, it's important to have a visual separation between the two split images.

The manual implementation here is fairly simple I think.

huchenlei replied:
Thanks for sharing the reason!

huchenlei reviewed (changes requested):
Can we compress these thumbnail preview images? Convert them to jpeg and reduce the dimension.

christian-byrne committed:
compress, convert webp, strip metadata, reduce dims

christian-byrne committed:
update coreTemplates to webp

christian-byrne commented:
Let me compress video as well

huchenlei reviewed (changes requested):
https://github.com/user-attachments/assets/0b0cf338-a33d-45ac-ae82-b46494c1edc5

It seems like clicking the thumbnail image no longer loads the workflow though.

christian-byrne commented:
Yes, I moved the interaction to the title to match the other designs I was referencing. It was also necessary because some of the images are interactable via the image compare slider. I can make all non-slider thumbnails open on click but it will be inconsistent. WDYT?

christian-byrne committed:
increase compression controlnet images

huchenlei commented:
> Yes, I moved the interaction to the title to match the other designs I was referencing. It was also necessary because some of the images are interactable via the image compare slider. I can make all non-slider thumbnails open on click but it will be inconsistent. WDYT?

Shall we make it more obvious on where to click?

Here are my suggestions
- On hovering we highlight the title clickable area
- Replace the title plain text with button

christian-byrne committed:
forbid dragging thumbnail to avoid confusion

christian-byrne committed:
use button for subtitle

christian-byrne commented:
I converted title to button and added call to action icon

![Selection_975](https://github.com/user-attachments/assets/6bca2f7a-58c4-4a26-997c-ab3e478855e2)

christian-byrne committed:
make thumbnail clickable, change slider to hover

christian-byrne committed:
update test fixture to click image
